syntax = "proto2";

package apollo.planning;

/////////////////////////////////////////////////////////////////
// The start_s and end_s are longitudinal values.
// start_s <= end_s.
//
//              end_s
//                ^
//                |
//          S  direction
//                |
//            start_s
//
// The start_l and end_l are lateral values.
// start_l <= end_l. Left side of the reference line is positive,
// and right side of the reference line is negative.
//  end_l  <-----L direction---- start_l
/////////////////////////////////////////////////////////////////
// s代表长度， l代表宽度, frenet坐标系中的boundary设置
message SLBoundary {                              // sl的边框由两个轴的终点和起点构成
  optional double start_s = 1;                    // s路程方向的起点
  optional double end_s = 2;                      // s路程方向的终点
  optional double start_l = 3;                    // l横向(宽度)方向上的起点
  optional double end_l = 4;                      // l横向(宽度)方向上的终点
}
